cirrus-ci_artifacts/cirrus-ci_artifacts.py

```diff
@@ -58,109 +61,91 @@ def get_raw_taskinfo(gqlclient, build_id):
-
-async def download_artifacts(task_name, art_names_urls, path_rx=None):
-    """Download artifact if path_rx unset or matches dest. path into CWD subdirs"""
-    async with aiohttp.ClientSession() as session:
-        for art_name, art_url in art_names_urls:
-            # Cirrus-CI Always/Only archives artifacts path one-level deep
-            # (i.e. no subdirectories).  The artifact name and filename were
-            # are part of the URL, so must decode them. See art_to_url() above
-            dest_path = join(task_name, unquote(art_name), basename(unquote(art_url)))
+    # Last path component assumed to be the filename
+    makedirs(split(dest_path)[0], exist_ok=True)  # os.path.split
+    async with session.get(dl_url) as response:
+        with open(dest_path, "wb") as dest_file:
+            dest_file.write(await response.read())
+
+async def download_artifacts(task, path_rx=None):
+    """Given a task dict, download all artifacts or matches to path_rx"""
+    downloaded=[]
+    skipped=[]
+    async with ClientSession() as session:
+        for art_url_sfx in task_art_url_sfxs(task):
+            dest_path = unquote(art_url_sfx)  # Strip off URL encoding
+            dl_url = f"{CCI_ART_URL}/{dest_path}"
             if path_rx is None or bool(path_rx.search(dest_path)):
-                print(f"Downloading '{dest_path}'")
-                sys.stderr.flush()
-                makedirs(dirname(dest_path), exist_ok=True)
-                with open(dest_path, "wb") as dest_file:
-                    data = await download_artifact(session, art_url)
-                    dest_file.write(data)
-
+                if VERBOSE:
+                    print(f"    Downloading '{dest_path}'")
+                    sys.stderr.flush()
+                await download_artifact(session, dest_path, dl_url)
+                downloaded.append(dest_path)
+            else:
+                if VERBOSE:
+                    print(f"       Skipping '{dest_path}'")
+                skipped.append(dest_path)
+    return {"downloaded":downloaded, "skipped":skipped}
 
 def get_arg(index, name):
-    """Return the value of command-line argument, raise error ref: name if empty"""
-    err_msg=f"Error: Missing/empty {name} argument\n\nUsage: {sys.argv[0]} <repo. owner/name> <bucket> <build ID> [path rx]"
+    """Return the value of command-line argument, raise error if index 0 empty"""
+    err_msg=f"Error: Missing/empty {name} argument\n\nUsage: {sys.argv[0]} [--verbose] <build ID> [path rx]"
     try:
-        result=sys.argv[index]
-        if bool(result):
+        result=sys.argv[index]  # Could raise IndexError
+        if bool(result.strip()):
             return result
-        else:
-            raise ValueError(err_msg)
+        raise IndexError
     except IndexError:
-        sys.stderr.write(f'{err_msg}\n')
-        sys.exit(1)
-
+        if index == 1:  # Arg 1 is required
+            raise ValueError(err_msg)
+        return None
```

Comment:
I'm growing very grumpy of this `get_arg()`, I'm going to throw in the towel and just use `argparse` like I should have done from the start :blush: